app/assets/javascripts/discourse/app/components/user-nav/dropdown-list.js

```diff
@@ -0,0 +1,54 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class UserNavDropdownList extends Component {
+  @tracked displayList = false;
+
+  get chevron() {
+    return this.displayList ? "chevron-up" : "chevron-down";
+  }
+
+  get defaultButtonClass() {
+    return "user-nav-dropdown-button";
+  }
+
+  get buttonClass() {
+    const props = [this.defaultButtonClass];
+
+    if (this.args.isActive) {
+      props.push("active");
+    }
+
+    return props.join(" ");
+  }
+
+  @action
+  toggleList() {
+    this.displayList = !this.displayList;
+  }
+
+  @bind
+  collapseList(e) {
```

Comment:
Very minor, IMO we should use `event` over `e` but it's also not a hill I will die on so feel free to ignore

```suggestion
  collapseList(event) {
```

Reply:
Done :+1: